tests/test-files/good/Elm-0.19/AllSyntax/Expressions.elm

```diff
@@ -273,6 +273,19 @@ lambdaWithMultilinePattern =
 ifStatement =
     let
         a =
+            if True then 1 else 2
+
+        b =
+            if {- B -} True then 1 else 2
```

Comment:
As said in a comment, `if {- A -} True then {- B -} 1 else {- C -} 2` is not accepted on single-line, should it be?

Reply:
Yeah, I think it makes sense for
```
if {- A -} True {- B -} then {- C -} 1 {- D -} else {- E -} 2
```
to stay on one line.  This can probably be done in your logic by using `formatCommented`, `formatPreCommented`, `formatPostCommented` to make a box that includes the comments first before destructuring to check for SingleLine.

Reply:
It was a little more complicated in order to avoid rendering twice the `then` and `else` bodies, I have therefore switched a top level `formatIfExpression` function.

Please tell me if you see some desirable formatting or code/splitting improvements.